atmospheric_explorer/plotting_apis_new.py

```diff
@@ -8,36 +8,122 @@
+    # pylint: disable=too-few-public-methods
 
     def __init__(
-        self: PlottingInterface,
+        self: EAC4Parameters,
         _data_variable: str,
         _var_name: str,
-        _time_period: str,
-    ):
+        _file_format: str,
+        _dates_range: str,
+        _time_values: str,
+    ) -> None:
         self.data_variable = _data_variable
         self.var_name = _var_name
-        self.time_period = _time_period
+        self.file_format = _file_format
+        self.dates_range = _dates_range
+        self.time_values = _time_values
 
-    @property
-    def data_variable(self: PlottingInterface) -> str:
-        """Data variable name"""
-        return self._data_variable
 
-    @data_variable.setter
-    def data_variable(
-        self: PlottingInterface,
-        data_variable_input: str,
+class GHGParameters:
+    """Parameters for GHG dataset"""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(
+        self: GHGParameters,
+        _data_variable: str,
+        _file_format: str,
+        _quantity: str,
+        _input_observations: str,
+        _time_aggregation: str,
+        _year: list[str],
+        _month: list[str],
     ) -> None:
-        self._data_variable = data_variable_input
+        self.data_variable = _data_variable
+        self.file_format = _file_format
+        self.quantity = _quantity
+        self.input_observations = _input_observations
+        self.time_aggregation = _time_aggregation
+        self.year = _year
+        self.month = _month
+
+
+class PlottingInterface(ABC):
+    """Generic interface for all plotting APIs"""
+
+    _plot_type: PlotType
+
+    def __init__(
+        self: PlottingInterface,
+        _dataset_name: DatasetName,
+        _eac4_parameters: EAC4Parameters,
+        _ghg_parameters: GHGParameters,
+        _countries: list[str],
+        _title: str,
+    ):
+        self.dataset_name = _dataset_name
+        self.title = _title
+        self.countries = _countries
+        # Qui sotto: puo` avere senso tenere comunque degli argomenti generici,
+        # che ci sono per entrambi i dataset e possono servire sempre nei plot/dowload dati?
+        match self.dataset_name:
+            case DatasetName.eac4:
+                self.eac4_parameters = _eac4_parameters
+                self.data_variable = _eac4_parameters.data_variable
+                self.var_name = _eac4_parameters.var_name
+            case DatasetName.ghg:
+                self.ghg_parameters = _ghg_parameters
+                self.data_variable = _ghg_parameters.data_variable
+                self.var_name = _ghg_parameters.var_name
+
+    @abstractmethod
+    def download_data(self: PlottingInterface):
+        """Downloads data"""
+        match self.dataset_name:
+            case DatasetName.eac4:
+                assert self.eac4_parameters is not None
+                data = EAC4Instance(
+                    self.eac4_parameters.data_variable,
```

Comment:
I think it would make sense if we used named args here as well, in the `EAC4Instance` class. I'd also suggest we make these class names more similar (e.g. EAC4Instance and GHGInstance?)